src/robot/variables/tablesetter.py

```diff
@@ -152,7 +186,12 @@ def _get_value_and_separator(self, value, separator):
     def _replace_variables(self, variables):
         value, separator = self.value, self.separator
         if self._is_single_value(value, separator):
-            return variables.replace_scalar(value[0])
+            replace_scalar = variables.replace_scalar
+            if self._is_secret_type():
+                value = self._handle_secrets(value[0], replace_scalar)
+            else:
+                value = value[0]
+            return replace_scalar(value)
```

Comment:
`_handle_secrets` already resolves variables, so the value returned it should be returned directly. The value it returns it always an object (other alternative is an exception) and `replace_scalar` does nothing to objects so this doesn't affect functionality, but the code gets simpler that way. It seems that at least list variable code has the same issue.